clang/test/Sema/types.c

```diff
@@ -9,20 +9,20 @@ typedef int (*T)[2];
 restrict T x;
 
 typedef int *S[2];
-restrict S y; // expected-error {{restrict requires a pointer or reference ('S' (aka 'int *[2]') is invalid)}}
-
-
+restrict S y;
```

Comment:
Can you also add a test for multidimensional arrays, like:
```
typedef int *S[2][2];
restrict S y;
```
and (this should already work because it's adding the restrict to a pointer to an array):
```
typedef int (*S)[2];
restrict S y;
```
and finally, can you add some type verification tests like (for each of the test cases):
```
static_assert(_Generic(typeof(y), int * restrict[2]: 1, default : 0));
```

Reply:
@AaronBallman thanks for the feedback. I've added additional tests